examples/deepspeed/trainer/model_def.py

```diff
@@ -0,0 +1,214 @@
+import logging
+from typing import Any, Dict, Iterator, Optional, Tuple, Union, cast
+
+import data
+import deepspeed
+import torch
+import torch.nn as nn
+import torch.utils.data
+import torchvision
+from attrdict import AttrDict
+from gan_model import Discriminator, Generator, weights_init
+
+from determined.pytorch import DataLoader, TorchData
+from determined.pytorch.deepspeed import (
+    DeepSpeedTrial,
+    DeepSpeedTrialContext,
+    overwrite_deepspeed_config,
+)
+
+REAL_LABEL = 1
+FAKE_LABEL = 0
+
+
+class DCGANTrial(DeepSpeedTrial):
+    def __init__(self, context: DeepSpeedTrialContext,
+                 hparams: AttrDict, data_config: AttrDict) -> None:
```

Comment:
feel like `AttrDict` is kinda unnecessary here, just make it a plain dict?

Reply:
Removed all traces of it from this example.